src/rf/apps/core/models.py

```diff
@@ -133,6 +133,10 @@ class Meta:
     thumb_large_key = CharField(max_length=255, blank=True, default='',
                                 help_text='S3 key for large thumbnail')
 
+    def has_copied_images(self):
+        image = self.layer_images.first()
```

Comment:
Just tested this, and it does in fact work with two origins. Error messages might be a little murky in some cases but in a successful case it looks like it will work.